common/djangoapps/util/views.py

```diff
@@ -90,6 +90,43 @@ def inner(request, *args, **kwargs):
     return decorator
 
 
+def conditional_decorator(decorator, test_func):
```

Comment:
@saleem-latif I think instead of writing a new decorator you should modify `handle_500` to raise exception if given condition is not met.